xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererBase.cpp

```diff
@@ -140,13 +140,17 @@ CRendererBase::CRendererBase(CVideoSettings& videoSettings)
 
 CRendererBase::~CRendererBase()
 {
-  if (DX::Windowing()->IsHDROutput())
+  // At stop playback restores Windows HDR state to previous state
+  // User preference Windows HDR on/off for desktop / Kodi GUI
+  if (m_AutoSwitchHDR)
   {
-    CLog::LogF(LOGDEBUG, "Restoring SDR rendering");
-    DX::Windowing()->SetHdrColorSpace(DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709);
-    if (m_AutoSwitchHDR)
-      DX::Windowing()->ToggleHDR(); // Toggle display HDR OFF
```

Comment:
This will toggle HDR if AutoSwitch is on whether or not the content being played is HDR. This could result in HDR getting toggled off and then back on when the next lines are run. Any chance we could use my version of this code in #23500 ? 

`if (m_AutoSwitchHDR)
  {
    if (m_initiallyHdrEnabled)
    {
      if (!DX::Windowing()->IsHDROutput())
	  {
	    CLog::LogF(LOGINFO, "Restoring HDR rendering (AutoSwitch)");
       	    DX::Windowing()->ToggleHDR(); // Toggle display HDR ON
           DX::Windowing()->SetHdrColorSpace(DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709);
	  }
	}
	else if (DX::Windowing()->IsHDROutput())
    {
      CLog::LogF(LOGINFO, "Restoring SDR rendering (AutoSwitch)");
      DX::Windowing()->ToggleHDR(); // Toggle display HDR OFF
      DX::Windowing()->SetHdrColorSpace(DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709);    
   }
  }`

This way checks first if autoswitch is enabled, then checks if HDR switch was enabled, then checks if output is HDR. No extra toggle would happen

Reply:
No, you are wrong.  This code only toggles HDR (one time) when current HDR state is different of initial HDR state (before start playback).

Explain how this code can toggle HDR two times:

```c++
  if (m_AutoSwitchHDR)
  {
    if (m_initiallyHdrEnabled != DX::Windowing()->IsHDROutput())
    {
      CLog::LogF(LOGDEBUG, "Restoring {} rendering", m_initiallyHdrEnabled ? "HDR" : "SDR");
      DX::Windowing()->ToggleHDR();
    }
  }
```

Reply:
It looks better now :) code was different yesterday when I looked